### PR TITLE
[backend] Use an unified CLI

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change simplifies the command API by providing an "hyper" tool
with sub commands:

- monocle-api -> monocle api
- macroscope -> monocle crawler
- lentille -> monocle lentille

Then the janitor can be implemented as another sub command.

### CODE REVIEW

File: deployment/api-deployment.yaml [hunk 1]
```diff
@@ -23,9 +23,8 @@ spec:
     spec:
       containers:
         - args:
-            - monocle-api
-            - --port
-            - "9898"
+            - monocle
+            - api
```
Comment (morucci) [inline deployment/api-deployment.yaml:26]: Looks like MONOCLE_API_PORT is missing from the env and we rely on the default in Monocle.CLI.

File: haskell/src/CLI.hs [hunk 2]
```diff
@@ -3,80 +3,86 @@
...
+
+    -- The janitor entrypoint
+    usageJanitor = pure $ putStrLn "NotImplemented"
+
+    -- Helper to create sub command
+    mkCommand doc name parser = command name $ info parser $ progDesc doc
+
+    -- Helpers to get value fromm the env
+    getFromEnv = Env.parse (header "monocle")
+    envConf = var str "CONFIG" (help "The Monocle configuration" <> def "/etc/monocle/config.yaml" <> helpDef show)
```
Comment (morucci) [inline haskell/src/CLI.hs:65]: I'm not able to discover the environment variables with for instance "cabal run exe:monocle -- api --help"

File: haskell/src/CLI.hs [hunk 3]
```diff
@@ -3,80 +3,86 @@
...
+    envMonitoring = var str "MONOCLE_CRAWLER_MONITORING" (help "The Monitoring Port" <> def "9001")
+    getInt txt = fromMaybe (error . from $ "Invalid number: " <> txt) $ readMaybe txt
+
+main :: IO ()
+main = join $ execParser opts
+  where
+    opts =
+      info
+        (usage <**> helper)
+        (fullDesc <> progDesc "change-metrics.io | monocle")
```
Comment (morucci) [inline haskell/src/CLI.hs:78]: changemetrics.io

File: haskell/src/CLI.hs (hunk 2 above)
Comment (TristanCacqueray) [reply haskell/src/CLI.hs:69]: Good idea, using https://hackage.haskell.org/package/optparse-applicative-0.16.1.0/docs/Options-Applicative.html#v:footer we can add extra doc, and we can generate the help from envparse with: https://hackage.haskell.org/package/envparse-0.4.1/docs/Env.html#v:helpDoc

**Review by morucci — APPROVED** [review]

**Comment by morucci** [comment]
hlint: https://github.com/change-metrics/monocle/runs/4355250065?check_suite_focus=true#step:5:1382

File: deployment/api-deployment.yaml (hunk 1 above)
Comment (TristanCacqueray) [reply deployment/api-deployment.yaml:27]: I think that's ok because this is in kubernetes internal network.

**Pull request merged**